### PR TITLE
Feature/authorization

### PR DESCRIPTION
**Pull request by svtslv17**
Made a connection to a remote server using the POST method. The received token is written to the Cookies. The isAuthorized flag is set in the store.

"email": "maxon@gmail.com",
"password": "12345BestPassword!"

"email": "user@example.com",
"password": "12345Ab!"

"email": "admin@gmail.com",
"password": "32131Adw$"

"email": "li@gmail.com",
"password": "136584Cw@"

### CODE REVIEW

File: src/components/header/headerNav/HeaderNav.jsx [hunk 1]
```diff
@@ -21,4 +24,8 @@ class HeaderNav extends React.Component {
   }
 }
 
-export default HeaderNav;
+const mapDispatchToProps = (dispatch) => ({
+  LogOut: () => dispatch(userLogOut()),
```
Comment (nsukhotskaya) [inline src/components/header/headerNav/HeaderNav.jsx:28]: Why did you use PascaleCase?

File: src/pages/login/Login.jsx [hunk 2]
```diff
@@ -1,13 +1,31 @@
...
 import './Login.sass';
 import { Footer } from '../../components';
 import { useMediaDown } from '../../components/utils';
 import { getFieldLabel } from '../../utils';
 
-const Login = () => {
+const Login = (props) => {
+  const { message } = props;
   const smallScreen = useMediaDown('sm');
+  const state = {
```
Comment (nsukhotskaya) [inline src/pages/login/Login.jsx:14]: Please read about useSate hook.
https://reactjs.org/docs/hooks-state.html

File: src/pages/login/Login.jsx [hunk 3]
```diff
@@ -1,13 +1,31 @@
...
+    email: '',
+    password: '',
+  };
+
+  const handleChange = (event) => {
+    const { name, value } = event.target;
+    state[name] = value;
```
Comment (nsukhotskaya) [inline src/pages/login/Login.jsx:21]: I'll pretend that I have not seen this line

File: src/pages/login/Login.jsx [hunk 4]
```diff
@@ -1,57 +1,109 @@
...
 
 import './Login.sass';
-
 import { Footer } from '../../components';
 import { useMediaDown } from '../../components/utils';
 import { getFieldLabel } from '../../utils';
 
-const Login = () => {
+const Login = (props) => {
+  const { message } = props;
```
Comment (itsnotpolza) [inline src/pages/login/Login.jsx:20]:
```suggestion
const Login = ({ message }) => {
```

File: src/pages/login/Login.jsx [hunk 5]
```diff
@@ -1,57 +1,109 @@
...
   const smallScreen = useMediaDown('sm');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+
+  const handleChange = (event) => {
+    const { name, value } = event.target;
```
Comment (itsnotpolza) [inline src/pages/login/Login.jsx:26]: you can manage email and password in one state object and handle change in a smaller function. this is only a suggestion for refactoring

File: src/store/actions/userLogIn.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { logInStarted, logInFailure, logInSuccess } from '../commands/types';
+import API from '../commands/api';
+import { history } from '../store';
+
+const userLogIn = (user) => async (dispatch) => {
+  dispatch(logInStarted());
```
Comment (itsnotpolza) [inline src/store/actions/userLogIn.js:6]: rename to loginRequest

File: src/store/commands/types.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+const logInSuccess = () => ({
+  type: 'LOG_IN_SUCCESS',
+  isAuthorized: true,
+});
+const logInFailure = () => ({
+  message: 'You entered incorrect data',
+  type: 'LOG_IN_FAILURE',
+});
+const logInStarted = () => ({
+  type: 'LOG_IN_STARTED',
```
Comment (itsnotpolza) [inline src/store/commands/types.js:10]: LOG_IN_REQUEST
Also keep action names in a constant object.

File: src/store/reducers/isAuthorized.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+const initialState = {
+  isAuthorized: false,
+};
+
+export default function isAuthorized(state = initialState, action) {
+  switch (action.type) {
+    case 'LOG_IN_STARTED':
```
Comment (itsnotpolza) [inline src/store/reducers/isAuthorized.js:7]:
```suggestion
    case 'LOG_IN_REQUEST':
```

File: src/pages/login/Login.jsx [hunk 9]
```diff
@@ -1,57 +1,102 @@
...
+import { userLogIn } from '../../store/actions';
 
 import './Login.sass';
-
 import { Footer } from '../../components';
 import { useMediaDown } from '../../components/utils';
 import { getFieldLabel } from '../../utils';
 
-const Login = () => {
+const Login = (state) => {
```
Comment (nsukhotskaya) [inline src/pages/login/Login.jsx:19]:
```suggestion
const Login = (props) => {
```

File: src/store/actions/userLogIn.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+};
+
+const userLogIn = (user) => async (dispatch) => {
+  dispatch(loginRequest);
+  try {
+    const response = await API.post(`api/authenticate`, user);
+    const token = response.data.accessToken;
+    localStorage.setItem('accessToken', token);
+    dispatch(loginSuccess);
+    history.push('/');
```
Comment (nsukhotskaya) [inline src/store/actions/userLogIn.js:22]: history.push('/') or
 dispatch(push("/") ?  (push from connected-react-router)

**Review by nsukhotskaya — APPROVED** [review]

**Review by ValeryiaMIRON — APPROVED** [review]

**Review by itsnotpolza — APPROVED** [review]

**Pull request merged**